frontend/src/main/scala/bloop/dap/DebugSessionLogger.scala

```diff
@@ -0,0 +1,63 @@
+package bloop.dap
+
+import java.net.InetSocketAddress
+
+import bloop.logging.{DebugFilter, Logger}
+import com.microsoft.java.debug.core.protocol.Events.OutputEvent
+
+import scala.concurrent.Promise
+
+/**
+ * Defines a logger that forwards some events to a debug session. Operations
+ * not forwarded to the session are passed onto the underlying logger.
+ *
+ * A key responsibility of this logger is to intercept the start JDI log and
+ * parse the debugging port of the remote machine. This port is then passed to
+ * bind the host and remote machines and run the JDI infrastructure.
+ */
+final class DebugSessionLogger(
+    debugSession: DebugSession,
+    listener: InetSocketAddress => Unit,
+    underlying: Logger
+) extends Logger {
+  @volatile private var initialized = false
+  override val name: String = s"${underlying.name}-debug"
+
+  override def isVerbose: Boolean = underlying.isVerbose
+  override def trace(t: Throwable): Unit = underlying.trace(t)
+  override def printDebug(msg: String): Unit = underlying.debug(msg)(DebugFilter.All)
+  override def warn(msg: String): Unit = underlying.warn(msg)
```

Comment:
In which scenarios could we be calling `warn`, `trace` or `debug` on `DebugSessionLogger`? Or are we somehow ensured these will never be called? Currently we pass them to the underlying logger but we should make a good effort to give clients as much feedback as possible.

Reply:
> Maybe we should forward all of the log levels to the underlying in addition to sending some of them to the DAP-client? Just a thought.
  
That's a great thought, we should definitely be doing that.